packages/dynamic-links/android/src/main/java/io/invertase/firebase/dynamiclinks/ReactNativeFirebaseDynamicLinksModule.java

```diff
@@ -144,6 +144,34 @@ public void getInitialLink(Promise promise) {
       });
   }
 
+  @ReactMethod
+  public void resolveLink(String link, Promise promise) {
+    try {
+      FirebaseDynamicLinks.getInstance()
+        .getDynamicLink(Uri.parse(link))
+        .addOnCompleteListener(task -> {
+          if (task.isSuccessful()) {
+            PendingDynamicLinkData linkData = task.getResult();
+            // Careful: link == null if link invalid, isSuccessful is only false on processing error
+            if (linkData != null && linkData.getLink() != null && linkData.getLink().toString() != null) {
+              String linkUrl = linkData.getLink().toString();
+              // I can't believe they made this an int when versionCode is moving to long, but okay
```

Comment:
```suggestion
```

Reply:
Yeah this was definitely a dev note - now at least you know that you'll see this line of code again though, when they move to long